lib/engine/part/node.rb

```diff
@@ -36,7 +36,19 @@ def select(paths, corporation: nil)
         on.keys.select { |p| on[p] == 1 }
       end
 
-      def walk(visited: nil, on: nil, corporation: nil, visited_paths: {}, skip_track: nil)
+      #
+      # Explore the paths and nodes reachable from this node
+      #
+      #
+      # visited: a hashset of visited Nodes
+      # visited_paths: a hashset of visited Paths
+      # visited_hexedges: a hashset of crossed HexEdges (2-size Arr of hex and edge to represent hex edge crossings)
+      # on: see Path::Walk
+      # corporation: If set don't walk on adjacent nodes which are blocked for the passed corporation
+      # skip_track: If passed, don't walk on track of that type (ie: :broad track for 1873)
+      #
+      # This method recursively bubbles up yielded values from nested Node::Walk and Path::Walk calls
+      def walk(visited: nil, on: nil, corporation: nil, visited_paths: {}, visited_hexedges: {}, skip_track: nil)
```

Comment:
visited_edges

Reply:
I've changed this and it now only stores 1 side of each edge crossed